src/ssvtools/modify_coordinates.py

```diff
@@ -0,0 +1,239 @@
+    result, buffer = srm.getBuffer()
+    sir = region.createStreaminformationRegion()
+    srm = sir.createStreamresourceMemoryBuffer(buffer)
+    result = region.read(sir)
+
+    del srm
+    del sir
+    del template_mesh3d
+    del template_fieldmodule
+    del template_region
+
+    derivative_xi1 = mesh3d.getChartDifferentialoperator(1, 1)
+    derivative_xi3 = mesh3d.getChartDifferentialoperator(1, 3)
+
+    elem_iter = mesh3d.createElementiterator()
+    element = elem_iter.next()
+    element_id = element.getIdentifier()
+    count = 0
+    while element.isValid():
+        eft = element.getElementfieldtemplate(coordinates_field, -1)
+        # Loop through coordinates field of trunk elements and set values to chosen SSV coordinates field. Also scale
+        # radius on template trunk to match relative radius in SSV trunk
+        if element_id <= elements_along_template_trunk:
+            if element_id == 1:
+                ln = [1, 2]
+            else:
+                ln = [2]
+            for i in range(len(ln)):
+                node = element.getNode(eft, ln[i])
+                fieldcache.setNode(node)
+                # get radius and rate of change of radius of template trunk
+                _, bx = tmp_coordinates_field.getNodeParameters(fieldcache, -1, Node.VALUE_LABEL_VALUE, 1, 3)
+                _, bd1 = tmp_coordinates_field.getNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D_DS1, 1, 3)
+                _, bd2 = tmp_coordinates_field.getNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D_DS2, 1, 3)
+                _, bd3 = tmp_coordinates_field.getNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D_DS3, 1, 3)
+                _, bd12 = tmp_coordinates_field.getNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D2_DS1DS2, 1, 3)
+                _, bd13 = tmp_coordinates_field.getNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D2_DS1DS3, 1, 3)
+                coordinates_field.setNodeParameters(fieldcache, -1, Node.VALUE_LABEL_VALUE, 1, bx)
+                coordinates_field.setNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D_DS1, 1, bd1)
+                coordinates_field.setNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D_DS2, 1,
+                                                    set_magnitude(bd2, unit_conversion_factor * bd2_mag_list[count]))
+                coordinates_field.setNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D_DS3, 1,
+                                                    set_magnitude(bd3, unit_conversion_factor * bd3_mag_list[count]))
+                coordinates_field.setNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D2_DS1DS2, 1,
+                                                    set_magnitude(bd12, unit_conversion_factor * bd12_mag_list[count]))
+                coordinates_field.setNodeParameters(fieldcache, -1, Node.VALUE_LABEL_D2_DS1DS3, 1,
+                                                    set_magnitude(bd13, unit_conversion_factor * bd13_mag_list[count]))
```

Comment:
As noted above, I don't the cross derivatives are going to be correct.
In particular, if the template is a straight line with constant radius then bd12 and bd13 will be zero vectors so set_magnitude won't work.